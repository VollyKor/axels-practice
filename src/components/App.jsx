import Container from "react-bootstrap/Container";
import { Switch, Route } from "react-router-dom";

import { Cart, Navbar } from "components";

import GlobalStyles from "../styled/globalStyles";

function App() {
  return (
    <>
      <GlobalStyles />
      <Navbar />
      <Switch>
        <Route path="/" exact>
          <Container className="p-3"></Container>
        </Route>
        <Route path="/cart">
          <Cart />
        </Route>
      </Switch>
    </>
  );
}

export default App;
