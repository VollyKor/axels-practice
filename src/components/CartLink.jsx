import { CartWrapper, CartText, CartNotification } from '../styled/CartLink';
import { IoCartOutline } from 'react-icons/io5';

const CartLink = ({ productsInCart = 0 }) => (
    <CartWrapper>
        <CartText className="mr-1">cart</CartText>
        <IoCartOutline size="25px" />
        <CartNotification className=" d-flex justify-content-center align-items-center">
            <span>{productsInCart}</span>
        </CartNotification>
    </CartWrapper>
);

export default CartLink;
