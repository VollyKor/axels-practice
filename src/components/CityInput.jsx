import { useState } from 'react';
import Form from 'react-bootstrap/Form';
import { BiTargetLock } from 'react-icons/bi';

import { FindButton } from '../styled/CityInput';
import GoogleModal from './GoodleModal';

export default function CityInput({
    inputRef,
    fillForm,
    value = '',
    ...props
}) {
    const [show, setShow] = useState(false);

    return (
        <div className="position-relative">
            <Form.Group controlId="city" className="position-relative">
                <Form.Control
                    type="text"
                    placeholder="City"
                    ref={inputRef}
                    value={value}
                    {...props}
                />
            </Form.Group>

            <FindButton type="button" onClick={() => setShow(!show)}>
                <BiTargetLock size="25px" color="#5a1094" />
            </FindButton>

            <GoogleModal
                fillForm={fillForm}
                show={show}
                onHide={() => setShow(false)}
            />
        </div>
    );
}
