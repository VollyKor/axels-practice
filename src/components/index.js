export { default as App } from "./App";
export { default as CartLink } from "./CartLink";
export { default as Cart } from "./Cart";
export { default as Navbar } from "./Navbar";
export { default as OrderNav } from "./OrderNav";
export { default as OrderSummary } from "./OrderSummary";
export { default as PriceCounter } from "./PriceCounter";
export { default as ShippingForm } from "./ShippingForm";
