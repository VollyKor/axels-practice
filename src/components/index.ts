export { default as App } from './App';
export { default as Cart } from './Cart';
export { default as CartLink } from './CartLink';
export { default as CityInput } from './CityInput';
export { default as Conditions } from './Conditions';
export { default as CreditCardIcon } from './CreditCardIcon';
export { default as GoodleModal } from './GoodleModal';
export { default as Navbar } from './Navbar';
export { default as NavBarLink } from './NavBarLink';
export { default as OrderData } from './OrderData';
export { default as OrderForms } from './OrderForms';
export { default as OrderNav } from './OrderNav';
export { default as OrderSummary } from './OrderSummary';
export { default as PriceCounter } from './PriceCounter';
export { default as PrintPage } from './PrintPage';
export { default as ProductImg } from './ProductImg';
export { default as ProductInfo } from './ProductInfo';
export { default as Tooltip } from './Tooltip';

// Forms
export { default as ShippingForm } from './forms/ShippingForm';
export { default as BillingForm } from './forms/BillingForm';
export { default as PaymentForm } from './forms/PaymentForm';
