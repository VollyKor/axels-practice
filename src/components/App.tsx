import Container from 'react-bootstrap/Container';
import { Switch, Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { Cart, Navbar } from 'components';
import PrintPage from './PrintPage';

import GlobalStyles from '../styled/utils/globalStyles';
import theme from '../styled/utils/themeStyles';
import TestCounter from './practice/TestCounter';
import { AnimateItem } from './practice/TestTransition';

const App = (): JSX.Element => (
    <ThemeProvider theme={theme}>
        <GlobalStyles />

        <Switch>
            <Route path="/print" exact>
                <PrintPage />
            </Route>
            <Route path="/">
                <Navbar />
                <Route path="/" exact>
                    <Container className="p-3">
                        <AnimateItem />
                        <TestCounter />
                    </Container>
                </Route>
                <Route path="/cart">
                    <Cart />
                </Route>
            </Route>
        </Switch>
    </ThemeProvider>
);

export default App;
