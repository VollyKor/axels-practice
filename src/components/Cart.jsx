import { Row, Col, Container } from "react-bootstrap";

import { ShippingForm, OrderNav, OrderSummary } from "components";

export default function Cart() {
  return (
    <Container>
      <Col lg={9} className="mx-auto ">
        <Row className="mx-auto bg-white shadow rounded">
          <Col lg={7} className="px-4">
            <OrderNav />
            <ShippingForm />
          </Col>
          <Col lg={5} className="position-relative bg-light rounded">
            <OrderSummary />
          </Col>
        </Row>
      </Col>
    </Container>
  );
}
