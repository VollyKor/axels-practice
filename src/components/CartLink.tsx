import { CartWrapper, CartText, CartNotification } from '../styled/CartLink';
import { IoCartOutline } from 'react-icons/io5';

interface Props {
    productsInCart?: number;
}

const CartLink = ({ productsInCart = 0 }: Props): JSX.Element => (
    <CartWrapper>
        <CartText className="mr-1">cart</CartText>
        <IoCartOutline size="25px" />
        <CartNotification className=" d-flex justify-content-center align-items-center">
            <span>{productsInCart}</span>
        </CartNotification>
    </CartWrapper>
);

export default CartLink;
