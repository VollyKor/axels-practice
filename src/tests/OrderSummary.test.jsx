import { shallow } from 'enzyme';
import * as redux from 'react-redux';

import { OrderSummary } from 'components';

import productsData from './utils/mockData';

describe('OrderSummary test', () => {
    let spyOnUseSelector;
    let spyOnUseDispatch;
    let mockDispatch;

    let initialProducts = null;

    beforeEach(() => {
        spyOnUseSelector = jest.spyOn(redux, 'useSelector');

        spyOnUseSelector.mockReturnValue(initialProducts);

        spyOnUseDispatch = jest.spyOn(redux, 'useDispatch');
        mockDispatch = jest.fn();
        spyOnUseDispatch.mockReturnValue(mockDispatch);
    });

    afterEach(() => {
        jest.restoreAllMocks();
    });

    it('render itself', () => {
        const wrapper = shallow(<OrderSummary />);
        expect(wrapper.exists()).toBe(true);
    });

    it('renders correctly with data', async () => {
        await spyOnUseSelector.mockReturnValue(productsData);

        const wrapper = shallow(<OrderSummary />);
        expect(wrapper).toMatchSnapshot();
    });
});
