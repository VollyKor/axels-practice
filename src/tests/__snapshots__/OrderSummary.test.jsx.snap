// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSummary test renders correctly with data 1`] = `
<Fragment>
  <Row
    className="justify-content-between pt-3 px-3 mt-2 align-items-baseline"
    noGutters={false}
  >
    <h2
      className="h6"
    >
      OrderSummary
    </h2>
    <withRouter(LinkContainer)
      to="/"
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        className="text-muted"
        disabled={false}
      >
        <small>
          <u>
            edit order
          </u>
        </small>
      </NavLink>
    </withRouter(LinkContainer)>
  </Row>
  <ListGroup
    className="list-group-flush"
  >
    <ListGroupItem
      active={false}
      className="border-top-3 p-2 pb-0 bg-transparent"
      disabled={false}
      key="132"
    >
      <Row
        noGutters={false}
      >
        <ProductImg
          img="https://pixabay.com/get/ge03278b9a4415afdfb3ba98ea3cc26558521c9ce7aa1419845348408127266ac67a221725aeba0c54f87bceabeb31eeb_640.jpg"
        />
        <ProductInfo
          color="black"
          name="Boots"
          quantity={1}
        />
        <styled.p>
          $
          100
        </styled.p>
      </Row>
    </ListGroupItem>
    <ListGroupItem
      active={false}
      className="border-top-3 p-2 pb-0 bg-transparent"
      disabled={false}
      key="146"
    >
      <Row
        noGutters={false}
      >
        <ProductImg
          img="https://pixabay.com/get/gf81b65afdc147703108c5b13db7759cc83582cdea7686ad840e67be5d06b4d3590400a8ccdfcd979caca9bc3bf9ac62a_640.jpg"
        />
        <ProductInfo
          color="brown"
          name="Shoes"
          quantity={1}
        />
        <styled.p>
          $
          130
        </styled.p>
      </Row>
    </ListGroupItem>
    <ListGroupItem
      active={false}
      className="border-top-3 p-2 pb-0 bg-transparent"
      disabled={false}
      key="173"
    >
      <Row
        noGutters={false}
      >
        <ProductImg
          img="https://pixabay.com/get/gbb7487e68df5a961adbe74bd167fcd7bf2e17cfb92ff2835becb7c6351bf37b04779e04238b805d9efa2b5a32fc6e076_640.jpg"
        />
        <ProductInfo
          color="gold"
          name="Sandals"
          quantity={1}
        />
        <styled.p>
          $
          70
        </styled.p>
      </Row>
    </ListGroupItem>
    <PriceCounter />
  </ListGroup>
</Fragment>
`;
